### DISCUSSION TITLE
If there's a way to truncate the db  because it has a lot of old data from pipelines that were deleted

### DISCUSSION THREAD

**Discussion by xaidu** [gocd/gocd#11795] [General]
I have a big DB of GOCD. I want to clean up some old data that is not used.

So do we have some tools to do it or something like  DB define doc to follow?

**Comment by chadlwilson** [comment]
Sadly, I am not aware of tools or approaches to handling DB cleanup, no. I have wondered myself, and at times wished I knew approaches to do this, and the consequences of doing so. There is a very old ticket at https://github.com/gocd/gocd/issues/879 that is not implemented.

The database schema itself is defined and migrated with Liquibase since `20.5.0` (see schema definition [here](https://github.com/gocd/gocd/tree/0f58107c851cf2df6ce7c6902eebde796dc1f742/db-support/db-migration/src/main/resources/db-migration-scripts))

Generally speaking, if people have a really large DB, it is because of the [MODIFICATIONS and MODIFIEDFILES tables](https://github.com/gocd/gocd/blob/0f58107c851cf2df6ce7c6902eebde796dc1f742/db-support/db-migration/src/main/resources/db-migration-scripts/initial/create-schema.xml#L193-L222), but I do not know the consequences of removing the data from just these tables (for old pipelines) _or_ also removing things from the related tables (PIPELINES, STAGES, BUILDS etc)

Also worth noting that there are not necessarily DB-level foreign key constraints that will stop you from deleting something that is needed, and clean-up in this way I don't believe has ever been officially supported to my knowledge.
